github/respository_rules_utils.go

```diff
@@ -0,0 +1,390 @@
+					RequiredDeploymentEnvironments: envs,
+				}
+
+				rulesSlice = append(rulesSlice, github.NewRequiredDeploymentsRule(params))
+			}
+		}
+	}
+
+	// Pattern parameter rules
+	for _, k := range []string{"commit_message_pattern", "commit_author_email_pattern", "committer_email_pattern", "branch_name_pattern", "tag_name_pattern"} {
+		if v, ok := rulesMap[k].([]interface{}); ok && len(v) != 0 {
+			patternParametersMap := v[0].(map[string]interface{})
+			if enabled, ok := patternParametersMap["enabled"].(bool); ok && enabled {
+
+				name := patternParametersMap["name"].(string)
+				negate := patternParametersMap["negate"].(bool)
+
+				params := &github.RulePatternParameters{
+					Name:     &name,
+					Negate:   &negate,
+					Operator: patternParametersMap["operator"].(string),
+					Pattern:  patternParametersMap["pattern"].(string),
+				}
+
+				switch k {
+				case "commit_message_pattern":
+					rulesSlice = append(rulesSlice, github.NewCommitMessagePatternRule(params))
+				case "commit_author_email_pattern":
+					rulesSlice = append(rulesSlice, github.NewCommitAuthorEmailPatternRule(params))
+				case "committer_email_pattern":
+					rulesSlice = append(rulesSlice, github.NewCommitterEmailPatternRule(params))
+				case "branch_name_pattern":
+					rulesSlice = append(rulesSlice, github.NewBranchNamePatternRule(params))
+				case "tag_name_pattern":
+					rulesSlice = append(rulesSlice, github.NewTagNamePatternRule(params))
+				}
+			}
+		}
+	}
+
+	// Pull request rule
+	if v, ok := rulesMap["pull_request"].([]interface{}); ok && len(v) != 0 {
+		pullRequestMap := v[0].(map[string]interface{})
+		if enabled, ok := pullRequestMap["enabled"].(bool); ok && enabled {
+			params := &github.PullRequestRuleParameters{
+				DismissStaleReviewsOnPush:      pullRequestMap["dismiss_stale_reviews_on_push"].(bool),
+				RequireCodeOwnerReview:         pullRequestMap["require_code_owner_review"].(bool),
+				RequireLastPushApproval:        pullRequestMap["require_last_push_approval"].(bool),
+				RequiredApprovingReviewCount:   pullRequestMap["required_approving_review_count"].(int),
+				RequiredReviewThreadResolution: pullRequestMap["required_review_thread_resolution"].(bool),
+			}
+
+			rulesSlice = append(rulesSlice, github.NewPullRequestRule(params))
+		}
+	}
+
+	// Required status checks rule
+	if v, ok := rulesMap["required_status_checks"].([]interface{}); ok && len(v) != 0 {
+		requiredStatusMap := v[0].(map[string]interface{})
+		if enabled, ok := requiredStatusMap["enabled"].(bool); ok && enabled {
+			requiredStatusChecks := make([]github.RuleRequiredStatusChecks, 0)
+			if requiredStatusChecksInput, ok := requiredStatusMap["required_check"].([]map[string]interface{}); ok {
+				for _, check := range requiredStatusChecksInput {
+					requiredStatusChecks = append(requiredStatusChecks, github.RuleRequiredStatusChecks{
+						Context:       check["context"].(string),
+						IntegrationID: check["integration_id"].(*int64),
```

Comment:
Could this by why it crashes? In the [example terraform](https://github.com/integrations/terraform-provider-github/pull/1808#issuecomment-1648126292) you only supplied `context`, and no `integration_id`. `integration_id` isn't required, but I think you need to only set `IntegrationID` based on if you actually have a value there

Reply:
Yeah that’s why alongside the way I was expanding/flattening this specific rule, I addressed those in my latest commit and I’ll update to add integration, we should also make `integration_id` required I think since I believe leaving it empty doesn’t actually set the rule correctly.

Reply:
We can't set the `integration_id` as required. I tested it a bit while doing my implementation, and I realized that local contexts, i.e., github actions that ran in my repository only had a context, and no integration_id. Example below

![image](https://github.com/integrations/terraform-provider-github/assets/34520175/b09ab491-a1ef-4186-8d88-b7b62b587efa)

Reply:
Oh that’s interesting, okay let’s keep it as is, maybe we should mention somewhere that if it’s an integration it needs an ID to actually be added.